Feature/fine tunning UI

FernandoLeonett opened the pull request:
refactor
- feature update project

I have started with retrofit

FernandoLeonett committed:
added btn save

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
ready third screen

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
init mvvm

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
init back button arrow

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
back button arrow

working ahi no mas

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
reservations screen is not working

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
working raw toolbar
Signed-off-by: Fernando <fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
working Ok

<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
improve callback delete button

<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
init interfaces
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
adds uses cases
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
im lost
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
working again
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
second screen is not working
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
second screen is  working
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

carohg25 commented on app/src/main/res/layout/fragment_lots.xml:
```diff
@@ -20,6 +20,7 @@
         android:layout_height="wrap_content"
         android:layout_marginStart="32dp"
```
I think you should use dimens folder to set dp values on all your xml files

FernandoLeonett committed:
working api
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
update pr
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

carohg25 commented on app/src/main/java/com/bootcamp_android/parking_app/view/fragments/LotsFragment.kt:
```diff
@@ -23,22 +23,20 @@ class LotsFragment : Fragment() {
-                layoutManager = LinearLayoutManager(activity)
-                adapter = LotsAdapter(it) { v -> lotClick(v) }
-            }
+        lotsViewModel.lots.observe(viewLifecycleOwner) { lots ->
```
You should observe the viewmodels on onViewCreated

carohg25 commented on app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationListFragment.kt:
```diff
@@ -14,43 +16,57 @@ import androidx.navigation.fragment.findNavController
 class ReservationListFragment : Fragment() {
 
     private lateinit var reservationsViewModel: ReservationsViewModel
+    private lateinit var lotsViewModel: LotsViewModel
```
You should use, one viewmodel in a fragment

carohg25 commented on app/src/main/java/com/bootcamp_android/parking_app/viewmodel/lot_detail/ReservationsViewModel.kt:
```diff
@@ -8,25 +10,22 @@ import com.bootcamp_android.domain.DeleteReservationUseCase
-    val successfulDelete: LiveData<Boolean> = mutableSuccessfulDelete
 
-    val successfulAdd: LiveData<Boolean> = mutableSuccessfulAdd
+    val mutableSuccessfulAdd = MutableLiveData<Boolean>()
```
You should use a private mutablelivedata and a public livedata, just like you did in LotViewModel

carohg25 commented on app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationListFragment.kt:
```diff
@@ -79,17 +97,20 @@ class ReservationListFragment : Fragment() {
 
+                if(reservationsViewModel.mutableSuccessfulDelete.value == true) {
 
-                if(reservationsViewModel.successfulDelete.value == true) {
```
You should **observe**, NOT call an attribute from its viewmodel class

carohg25 replied:
Remember if you want to get data from the viewmodel in the fragment, you should observe it

FernandoLeonett committed:
room is roompiendo
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
update pr room not working
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
the reservation appear again
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
fixing delete reservationdatabse
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
problem selecting the lot of spinner
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
problem selecting the lot of spinner
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
problem selecting the lot of spinner
<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
solvede the spinner problem now i have delete problem is broking

<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
solvede the spinner problem now i have delete problem is broking

<fernandoleonett@gmail.com>

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
Merge remote-tracking branch 'origin/feature/fine_tunning_UI' into feature/fine_tunning_UI

Signed-off-by: Fernando <fernandoleonett@gmail.com>

# Conflicts:
#	app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationAddFragment.kt
#	app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationListFragment.kt

FernandoLeonett committed:
add reservation working

Signed-off-by: Fernando <fernandoleonett@gmail.com>

# Conflicts:
#	app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationAddFragment.kt
#	app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationListFragment.kt

FernandoLeonett committed:
finish room

Signed-off-by: Fernando <fernandoleonett@gmail.com>

# Conflicts:
#	app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationAddFragment.kt
#	app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationListFragment.kt

FernandoLeonett committed:
added enum class for request

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
added string resources

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
added string resources

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
Merge remote-tracking branch 'origin/feature/fine_tunning_UI' into feature/fine_tunning_UI

Signed-off-by: Fernando <fernandoleonett@gmail.com>

# Conflicts:
#	app/src/main/java/com/bootcamp_android/parking_app/view/fragments/ReservationAddFragment.kt
#	app/src/main/res/layout/fragment_add_reservation.xml

FernandoLeonett committed:
starting testing

Signed-off-by: Fernando <fernandoleonett@gmail.com>

FernandoLeonett committed:
finish

Signed-off-by: Fernando <fernandoleonett@gmail.com>
